### PR TITLE
Sites Grid: Remove useless components and props

### PR DESCRIPTION
**Pull request by youknowriad**
## Proposed Changes

While working on some related changes to the sites dashboard, I noticed that the `SitesGrid` component and `SiteGridItem` are only used today in the site import flow in "select mode". Which means that the `siteSelectorMode` was always true. That prop hides a lot of UI from `SitesGrid` and `SiteGridItem`. So it's basically mostly dead code. This PR, removes the useless props and the dead code as a consequence.

## Testing Instructions

 - Test that this doesn't impact the Sites dashboard `/sites`  
 - Test that the site grid in the import flow still works properly `/setup/import-focused/sitePicker?siteSlug={slug}`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/sites-dashboard/components/test/sites-grid-item.tsx [hunk 1]
```diff
@@ -38,38 +38,11 @@ describe( '<SitesGridItem>', () => {
...
 				<SitesGridItem
 					site={ makeTestSite( { title: 'The example site' } ) as SiteExcerptData }
-					showLaunchNag={ false }
-					showBadgeSection={ false }
-					showThumbnailLink={ false }
-					showSiteRenewLink={ false }
+					onSiteSelectBtnClick={ () => {} }
 				/>
 			)
 			.toJSON();
```
Comment (tyxla) [inline client/sites-dashboard/components/test/sites-grid-item.tsx]: This test feels a bit useless 😅 I'm not sure what it's even attempting to test.

File: client/sites-dashboard/components/sites-grid-item/docs/example.jsx [hunk 2]
```diff
@@ -29,18 +29,24 @@ const itemClassName = css( {
...
 	return (
 		<div className={ clsx( container, className ) }>
 			{ sampleSiteData.map( ( site ) => (
 				<div className={ itemClassName } key={ site.ID }>
-					<SitesGridItem site={ site } key={ site.ID } />
+					<SitesGridItem
+						site={ site }
+						key={ site.ID }
+						onSiteSelectBtnClick={ () => {
+							// console.log( _site );
```
Comment (tyxla) [inline client/sites-dashboard/components/sites-grid-item/docs/example.jsx]: We might as well log it 🤷 .

**Review by tyxla — APPROVED** [review]
Great cleanup 👍 

I've verified the removed code is not in use.

🚀

File: client/sites-dashboard/components/test/sites-grid-item.tsx (hunk 1 above)
Comment (youknowriad) [reply client/sites-dashboard/components/test/sites-grid-item.tsx]: Agree, I'm just going to remove it I think :)

Comment (tyxla) [reply client/sites-dashboard/components/test/sites-grid-item.tsx]: Go for it 🧹

**Pull request merged**